src/sonic-host-services/scripts/hostcfgd

```diff
@@ -209,29 +369,78 @@ class AaaCfg(object):
                 servers_conf.append(server)
             servers_conf = sorted(servers_conf, key=lambda t: int(t['priority']), reverse=True)
 
+        radius_global = self.radius_global_default.copy()
+        radius_global.update(self.radius_global)
+
+        # RADIUS: Set the default nas_ip, and nas_id
+        if 'nas_ip' not in radius_global:
+            nas_ip = self.get_interface_ip("eth0")
+            if len(nas_ip) > 0:
+                radius_global['nas_ip'] = nas_ip
+        if 'nas_id' not in radius_global:
+            nas_id = self.get_hostname()
+            if len(nas_id) > 0:
+                radius_global['nas_id'] = nas_id
+
+        radsrvs_conf = []
+        if self.radius_servers:
+            for addr in self.radius_servers:
+                server = radius_global.copy()
+                server['ip'] = addr
+                server.update(self.radius_servers[addr])
+
+                if 'src_intf' in server:
+                    # RADIUS: Log a message if src_ip is already defined.
+                    if 'src_ip' in server:
+                        syslog.syslog(syslog.LOG_INFO, \
+            "RADIUS_SERVER|{}: src_intf found. May ignore src_ip".format(addr))
+                    # RADIUS: If server.src_intf, then get the corresponding
+                    # src_ip based on the server.ip, and set it.
+                    src_ip = self.get_interface_ip(server['src_intf'], addr)
+                    if len(src_ip) > 0:
+                        server['src_ip'] = src_ip
```

Comment:
I get that. My comment is, 
server['src_ip'] = src_ip, irrespective of len.
For e.g. len(server['src_ip'] > 0 && len(src_ip) == 0, don't we need to nullify server['src_ip'] entry ?

Reply:
I think I understand the comment as requesting the following. If the user specified both, a valid 'src_ip' at the global level, and a src_intf for the server, but the src_intf could not resolve to a valid IP address, nullify the 'src_ip' attribute of the server, since the user attempted to over-ride it with a src_intf (which could not resolve to a valid IP address). In other words, if a source interface was specified, ignore src-ip. I will fix with comment